openedx/core/djangoapps/content/course_overviews/signals.py

```diff
@@ -11,9 +11,10 @@
 def _listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=unused-argument
     """
     Catches the signal that a course has been published in Studio and
-    invalidates the corresponding CourseOverview cache entry if one exists.
+    updates the corresponding CourseOverview cache entry.
     """
     CourseOverview.objects.filter(id=course_key).delete()
+    CourseOverview.load_from_module_store(course_key)
```

Comment:
There's no need to trust my instincts -- we monitor how often course_structures fires off in New Relic. Looking at edx.org across the last week, I see an average of 1.2 rpm, with a 10 rpm spike. Edge averages 0.59 rpm with a 4 rpm spike. Definitely worth monitoring, but I don't think it's a cause for immediate concern.